src/snowflake/cli/api/project/definition_conversion.py

```diff
@@ -380,33 +450,118 @@ def _convert_templates_in_files(
+    # Set up fakers so that references to ctx.env. and fn.
+    # get templated to the same literal, since those references
+    # are the same in v1 and v2
+    replacement_template_context["ctx"]["env"] = _EnvFaker()
+    replacement_template_context["fn"] = _FnFaker()
+
     metrics = get_cli_context().metrics
-    metrics.set_counter_default(CLICounterField.PACKAGE_SCRIPTS, 0)
 
-    if (na := definition_v1.native_app) and (pkg := na.package) and pkg.scripts:
-        metrics.set_counter(CLICounterField.PACKAGE_SCRIPTS, 1)
-        cli_console.warning(
-            "WARNING: native_app.package.scripts is deprecated. Please migrate to using native_app.package.post_deploy."
-        )
-        # If the v1 definition has a Native App with a package, we know
+    if na := definition_v1.native_app:
+        # If the v1 definition has a Native App, we know
         # that the v2 definition will have exactly one application package entity
-        pkg_entity: ApplicationPackageEntityModel = list(
+        pkg_model: ApplicationPackageEntityModel = list(
             definition_v2.get_entities_by_type(
                 ApplicationPackageEntityModel.get_type()
             ).values()
         )[0]
-        converted_post_deploy_hooks = _convert_package_script_files(
-            project_root, pkg.scripts, pkg_entity, in_memory
-        )
-        if pkg_entity.meta is None:
-            pkg_entity.meta = MetaField()
-        if pkg_entity.meta.post_deploy is None:
-            pkg_entity.meta.post_deploy = []
-        pkg_entity.meta.post_deploy += converted_post_deploy_hooks
+
+        # Convert templates in artifacts by passing them through the TemplatesProcessor
+        # but providing a context that maps v1 template references to the equivalent v2
+        # references instead of resolving to literals
+        # For example, replacement_template_context might look like
+        # {
+        #     "ctx": {
+        #         "native_app": {
+        #             "bundle_root": "<% ctx.entities.pkg.bundle_root %>",
+        #             "deploy_root": "<% ctx.entities.pkg.deploy_root %>",
+        #             "application": {
+        #                 "name": "<% ctx.entities.app.identifier %>",
+        #             }
+        #             and so on...
+        #          }
+        #     }
+        # }
+        # We only convert files on-disk if the "templates" processor is used in the artifacts
+        # and if we're doing a permanent conversion. If we're doing an in-memory conversion,
+        # the CLI global template context is already populated with the v1 definition, so
+        # we don't want to convert the v1 template references in artifact files
+        if not in_memory and any(
+            processor.name == TEMPLATES_PROCESSOR
+            for artifact in pkg_model.artifacts
+            for processor in artifact.processors
+        ):
+            metrics.set_counter(CLICounterField.TEMPLATES_PROCESSOR, 1)
```

Comment:
I don't think I see anything that would set this to 0 in the case where we don't invoke the template processor in the conversion? In that case maybe I would turn the expression in the if statement into a boolean and pass that as 0/1 to set_counter

Reply:
yeah i could do that. i think i just copied this from the prod code

Reply:
In the other commands we have the set_counter_defaults in `compile_artifacts` which is called when bundling, but that doesn't happen here I don't think since we're manually building the bundle map etc.